pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +111,96 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoint reflects the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
+
+	// High level state string: Pending, Standby, Loading, Loaded, FailedToLoad
+	// +kubebuilder:default=Pending
+	ActiveModelState ModelState `json:"activeModelState"`
+	// +kubebuilder:default=""
+	TargetModelState ModelState `json:"targetModelState"`
+
+	// Details of last failure, when load of target model is failed or blocked
+	// +optional
+	LastFailureInfo *FailureInfo `json:"lastFailureInfo,omitempty"`
+
+	// How many copies of this predictor's models failed to load recently
+	// +kubebuilder:default=0
+	FailedCopies int `json:"failedCopies"`
```

Comment:
The intention was to have additional fields here `loadedCopies` and `loadingCopies`. Unfortunately it's not easy to distinguish between loaded and loading because we don't record that transition explicitly in etcd - but it will become possible with some internal improvements to model-mesh that we'd like to do at some point.

It would probably be good however to add a `totalCopies` field, that one would be easy to populate and probably still useful to have in the status.

Reply:
Sure, I think this would be useful. In SMS, this can correspond to the number of replicas, though would need to check how to populate this for both the Knative and raw deployment cases.